src/tests/Microsoft.Diagnostics.NETCore.Client/WriteDumpTests.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+using Xunit.Abstractions;
+
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.TestHelpers;
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class WriteDumpTests
+    {
+        private readonly ITestOutputHelper output;
+
+        public WriteDumpTests(ITestOutputHelper outputHelper)
+        {
+            output = outputHelper;
+        }
+
+        /// <summary>
+        /// A simple test that writes a single dump file
+        /// </summary>
+        [Fact]
+        public void BasicWriteDumpTest()
+        {
+            var dumpPath = "./myDump.dmp";
+            TestRunner runner = new TestRunner(CommonHelper.GetTraceePath(), output);
+            runner.Start(3000);
+            DiagnosticsClient client = new DiagnosticsClient(runner.Pid);
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => client.WriteDump(DumpType.Normal, dumpPath));
+            }
+            else
+            {
+                output.WriteLine($"Requesting dump at {DateTime.Now.ToString()}");
+                client.WriteDump(DumpType.Normal, dumpPath);
+                Assert.True(File.Exists(dumpPath));
+                File.Delete(dumpPath);
+            }
+            runner.Stop();
+        }
+
+        /// <summary>
+        /// A test that writes all the different types of dump file
+        /// </summary>
+        [Fact]
+        public void WriteAllDumpTypesTest()
```

Comment:
You might want to check with Mike if this test is valuable given other tests we already have that verify the content of different kinds of dumps.